File: node-sdk/examples/wrapper.js
/*
  wrapper.js
  SafeLM Node.js Example Wrapper
  
  Usage: Wrap this around your main application logic or Express server.
  The SafeLM SDK handles WAF, Token Compression, and Crash Prevention transparently!
*/

const SafeLM = require('../safelm');
const fs = require('fs');
const path = require('path');

// 1. Define your SafeLM License Key
const SAFELM_API_KEY = process.env.SAFELM_API_KEY || "safelm_sdk_demo_key";

async function runApplication() {
  console.log("[App] Initializing SafeLM Wrapper...");
  
  try {
    // 2. Wrap your application initialization with SafeLM
    // Provide the path to your config file if you have specific domains.
    await SafeLM.init(SAFELM_API_KEY, path.join(__dirname, '../safelm.config.json'));
    console.log("[App] SafeLM acts as an invisible shield for out/inbound traffic!\n");

    // 3. Your MAIN application code goes here!
    // SafeLM will transparently un-redact PII, compress prompts, block crashes, etc.
    
    // Example: Normal code fetching from an LLM. 
    // You don't need to change any logic inside!
    console.log("[App] Proceeding with main tasks...");
    
    // Simulate your SaaS logic:
    // const express = require('express');
    // const app = express();
    // app.listen(3000, () => console.log('SaaS running safely protected by SafeLM'));

  } catch (error) {
    console.error("[App] Application initialization failed: ", error);
    process.exit(1);
  }
}

runApplication();

File: python-sdk/examples/wrapper.py
# wrapper.py
# SafeLM Python Example Wrapper

import os
import sys

# 1. Import the SafeLM SDK
import safelm

# 2. Get your SafeLM License Key
SAFELM_API_KEY = os.environ.get("SAFELM_API_KEY", "safelm_sdk_demo_key")

def run_application():
    print("[App] Initializing SafeLM Wrapper...")
    
    try:
        # 3. Initialize the SDK Wrapper to catch crashes and apply protections
        # SafeLM seamlessly hooks into sys.excepthook, urllib.request, etc.
        safelm.init(SAFELM_API_KEY, config_path="../safelm.config.json")
        print("[App] SafeLM initialized! Application runs under a shield.")
        
        # 4. Main Application Loop
        # Insert your Flask, Django, or FastAPI runner here.
        # Everything fetched via urllib or parsed via JSON is filtered transparently.
        print("[App] Proceeding with main execution logic.")
        
        # For example, simulating a server:
        # app.run(port=8080)
        
    except Exception as e:
        print(f"[App] initialization failed: {e}")
        sys.exit(1)

if __name__ == "__main__":
    run_application()
